stan/math/prim/constraint/sum_to_zero_constrain.hpp

```diff
@@ -0,0 +1,142 @@
+#ifndef STAN_MATH_PRIM_CONSTRAINT_SUM_TO_ZERO_CONSTRAIN_HPP
+#define STAN_MATH_PRIM_CONSTRAINT_SUM_TO_ZERO_CONSTRAIN_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/fun/to_ref.hpp>
+#include <stan/math/prim/fun/inv_sqrt.hpp>
+#include <stan/math/prim/functor/apply_vector_unary.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return a vector with sum zero corresponding to the specified
+ * free vector.
+ *
+ * The sum-to-zero transform is defined using a modified version of the
+ * the inverse of the isometric log ratio transform (ILR).
+ * See:
+ * Egozcue, Juan Jose; Pawlowsky-Glahn, Vera; Mateu-Figueras, Gloria;
+ * Barcelo-Vidal, Carles (2003), "Isometric logratio transformations for
+ * compositional data analysis", Mathematical Geology, 35 (3): 279–300,
+ * doi:10.1023/A:1023818214614, S2CID 122844634
+ *
```

Comment:
For the ILR to "pivot-coordinates" this is described in 

```
Filzmoser P, Hron K, Templ M (2018). Applied Compositional Data Analysis. With Worked Examples in R. Springer International Publishing. Springer Nature Switzerland AG, Cham, Switzerland. ISBN 978-3-319-96420-1, [doi:10.1007/978-3-319-96422-5](https://doi.org/10.1007/978-3-319-96422-5), Springer Series in Statistics.
```

Reply:
Thanks!